base/dcp_client_metadata.go

```diff
@@ -196,17 +218,37 @@ func NewDCPMetadataCS(store DataStore, numVbuckets uint16, numWorkers int, keyPr
 	return m
 }
 
-func (m *DCPMetadataCS) Rollback(vbID uint16) {
+// OneshotRollback resets the metadata, preserving EndSeqNo if set, should be used if you want stream to end
+// after rollback is complete
+func (m *DCPMetadataCS) OneshotRollback(vbID uint16) {
 	// Preserve endSeqNo on rollback
 	endSeqNo := m.metadata[vbID].EndSeqNo
+	rollbackStartSeqno := findRollbackStartSeqno(m.metadata[vbID].FailoverEntries)
 	m.metadata[vbID] = DCPMetadata{
 		VbUUID:          0,
-		StartSeqNo:      0,
+		StartSeqNo:      rollbackStartSeqno,
 		EndSeqNo:        endSeqNo,
```

Comment:
When I look more at this, endSeqNo should be preserved ! If you see this getting reset to not  `gocbcore.SeqNo(0xffffffffffffffff)` we are writing that in an incorrect way somewhere. Then we wouldn't need a separate function for continuous and non continous? We only call `SeqEndSeqNos` in one place, opening a one shot stream. So maybe the actual problem has not yet been identified?

My best guess would be somehow we are re-initializing metadata on failover and see `EndSeqNo` as 0? But this is just because I don't have an idea on how this could happen otherwise.